backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,68 @@
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import Collection, CollectionMetadata, CollectionVersion, DatasetAsset, DatasetMetadata, DatasetStatus
+
+
+class DatabaseProviderInterface:
+    def create_collection(self, collection_metadata: CollectionMetadata):
+        pass
+
+    def get_collection(self, collection_id: str) -> Collection:
+        pass
+
+    def get_all_collections(self) -> List[Collection]:  # TODO: add filters if needed
+        pass
+
+    def delete_collection(self, collection_id: str) -> None:
+        pass
+
+    def update_collection_metadata(self, version_id: str, collection_metadata: CollectionMetadata) -> None:
+        pass
+
+    def update_collection_publisher_metadata(self, version_id: str, publisher_metadata: dict) -> None:
+        pass
+
+    def create_collection_version(self, collection_id: str) -> str:
+        pass
+
+    def delete_collection_version(self, collection_id: str, version_id: str) -> None:
```

Comment:
physical or logical deletion? is it even needed? logically, I think we just need to tombstone canonical Collections, rather than any particular version.  Versions will just be superseded by new versions.

Reply:
We do have the option to discard a revision, hence this method. The implementation might not be straightforward though.